subprojects/launcher/src/main/java/org/gradle/launcher/cli/CommandLineActionFactory.java

```diff
@@ -65,6 +67,8 @@
     private static final String HELP = "h";
     private static final String VERSION = "v";
 
+    private final ScriptFileResolver scriptFileResolver = DefaultScriptFileResolver.forLenientScriptingLanguages();
```

Comment:
❌  Since this seems to be the only call site requiring this `lenient` service behavior I wonder if there wouldn't be a more contained solution to this problem.

Reply:
I wonder too. We could introduce a `LenientServiceLocator` implementation to limit the changes in `DefaultServiceLocator`. These two would share a lot of code though. Any idea?